lib/trino-filesystem-manager/src/main/java/io/trino/filesystem/manager/FileSystemConfig.java

```diff
@@ -20,9 +20,9 @@
 public class FileSystemConfig
 {
     private boolean hadoopEnabled = true;
-    private boolean nativeAzureEnabled;
-    private boolean nativeS3Enabled;
-    private boolean nativeGcsEnabled;
+    private boolean nativeAzureEnabled = true;
```

Comment:
Switching all this on is very much a breaking change for users so we have to prepare for that with docs, upgrade guide and then give it quite some time for testing .. this PR now might be premature

Reply:
I am aware this is a breaking change, because we lost all the `hive.` prefixes from configs.
It seems we could make it much easier for users if we leveraged `@LegacyConfig`, but there might be some good reasons why we didn't do that.
Agreeing on the docs, but worth pointing out that making the change later won't make it less breaking.

Reply:
I am not sure but I think a clean break to new consistent property names across filesystems and a limited set it a good approach. The changes are so big that an upgrade would always only be partial and therefore also confusing to users. Might as well force them to really pay attention.

Reply:
>Might as well force them to really pay attention.

They won't say 'thank you' to us for having to review all the configs.
Especially in cases where we could trivially just accept existing configs and just issue deprecation warning.

Reply:
I don't think such an easy case exists often or is likely to be successful, and the cost of complexity to support clean and simple migration is large. I therefore vote against this. To me it feels like carrying all the cruft forward into the future for a one time update per user. However essentially that is a decision that we have to make as maintainers together. You know my vote..

Reply:
> I don't think such an easy case exists often

- authentication with instance profile, authentication with `~/.aws/credentials` (default authentication chain)
- explicit keys
- with or without specifying endpoint (S3 vs MinIO, etc.)

Basically all the changes https://github.com/trinodb/trino/pull/20639 could be made _after_ native FS is enabled, with just a bunch of `@LegacyConfig` annotations on the new config classes.

in https://github.com/trinodb/trino/pull/20639 I ommitted very few environments where update required more than just chopping off `hive.` from config prefix.  

>  To me it feels like carrying all the cruft forward into the future for a one time update per user.

What cruft would we be carrying?
We're saving arguably little effort of ours, serving big pain to users. Not an awesome deal.

> However essentially that is a decision that we have to make as maintainers together. 

if that be the case... cc @trinodb/maintainers .